### PR TITLE
Review errors.rst and fix consistency errors

### PR DESCRIPTION
**Pull request by aaroncohen73**
Made error names and descriptions more consistent

### CODE REVIEW

**Comment by MorganEPatch** [comment]
:+1:

File: source/errors.rst [hunk 1]
```diff
@@ -35,7 +35,7 @@ Slug) which does not match an object in the database.
 
   {
     "status": 404,
-    "error": "Object not found",
+    "error": "Object Not Found",
```
Comment (mathuin) [inline source/errors.rst]: https://github.com/osuosl/timesync-node/blob/develop/src/errors.js#L33 shows `'Object not found'` is the correct form.

File: source/errors.rst [hunk 2]
```diff
@@ -54,7 +54,7 @@ disclosing sensitive information.
 
   {
     "status": 500,
-    "error": "Server error",
+    "error": "Server Error",
```
Comment (mathuin) [inline source/errors.rst]: https://github.com/osuosl/timesync-node/blob/develop/src/errors.js#L44-L46 shows `'Server error'` is the correct form.

File: source/errors.rst [hunk 3]
```diff
@@ -72,7 +72,7 @@ which does not have a valid project.)
 
   {
     "status": 409,
-    "error": "Invalid foreign key",
+    "error": "Invalid Foreign Key",
```
Comment (mathuin) [inline source/errors.rst]: https://github.com/osuosl/timesync-node/blob/develop/src/errors.js#L58 shows `'Invalid foreign key'` is the correct form.

File: source/errors.rst [hunk 4]
```diff
@@ -138,7 +138,7 @@ With multiple invalid identifiers, the error is formatted like so:
 
   {
     "status": 400,
-    "error": "Invalid identifier",
+    "error": "Invalid Identifier",
```
Comment (mathuin) [inline source/errors.rst]: https://github.com/osuosl/timesync-node/blob/develop/src/errors.js#L138 shows `'The provided identifier was invalid'` as the correct form.

File: source/errors.rst [hunk 5]
```diff
@@ -155,7 +155,7 @@ username is provided.
 
   {
     "status": 401,
-    "error": "Invalid username",
+    "error": "Invalid Username",
```
Comment (mathuin) [inline source/errors.rst]: https://github.com/osuosl/timesync-node/blob/develop/src/errors.js#L138 shows `'Invalid username'` as the correct form.

File: source/errors.rst [hunk 6]
```diff
@@ -170,7 +170,7 @@ server is running.
 
   {
     "status": 401,
-    "error": "Authentication failure",
+    "error": "Authentication Failure",
```
Comment (mathuin) [inline source/errors.rst]: https://github.com/osuosl/timesync-node/blob/develop/src/errors.js#L154 shows `'Authentication failure'` as the correct form.

File: source/errors.rst [hunk 7]
```diff
@@ -186,7 +186,7 @@ contain a slug that already exists.
 
   {
     "status": 409,
-    "error": "Slug already exists",
+    "error": "Slug Already Exists",
```
Comment (mathuin) [inline source/errors.rst]: https://github.com/osuosl/timesync-node/blob/develop/src/errors.js#L169 shows `'The slug provided already exists'` as the correct form.

File: source/errors.rst [hunk 8]
```diff
@@ -90,17 +90,17 @@ duration field, or a project without a name.)
 
   unknown_field_error = {
     "status": 400,
-    "error": "Bad object",
+    "error": "Bad Object",
```
Comment (mathuin) [inline source/errors.rst]: https://github.com/osuosl/timesync-node/blob/develop/src/errors.js#L112 shows `'Bad object'` as the correct form.

File: source/errors.rst [hunk 9]
```diff
@@ -197,7 +197,7 @@ If multiple slugs are duplicated:
 
   {
     "status": 409,
-    "error": "Slugs already exist",
+    "error": "Slugs Already Exist",
```
Comment (mathuin) [inline source/errors.rst]: https://github.com/osuosl/timesync-node/blob/develop/src/errors.js#L169 shows the same string is used for single slugs that already exist and for multiple slugs that already exist.

File: source/errors.rst [hunk 10]
```diff
@@ -272,7 +272,7 @@ is already used by another user. Compare 8. Slug Already Exists.
 
   {
     "status": 409,
-    "error": "Username already exists",
-    "text": "username ${username} already exists",
+    "error": "Username Already Exists",
```
Comment (mathuin) [inline source/errors.rst]: https://github.com/osuosl/timesync-node/blob/develop/src/errors.js#L222 shows that `'Username already exists'` is the correct form.

File: source/errors.rst [hunk 11]
```diff
@@ -272,7 +272,7 @@ is already used by another user. Compare 8. Slug Already Exists.
...
+    "text": "Username ${username} already exists",
```
Comment (mathuin) [inline source/errors.rst]: The above line also shows that `'username ' + name + ' already exists'` is the correct form.

**Review by mathuin — CHANGES_REQUESTED** [review]
I agree entirely that osuosl/timesync-node's src/errors.js and this file have to have the same information.  Let's consider the API to be the single source of truth -- set this however you think is right and ask for review from @LyonesGamer and maybe someone else, then make a PR for osuosl/timesync-node to make that one match this one, and we should be done with this issue.

File: source/errors.rst [hunk 12]
```diff
@@ -55,7 +55,7 @@ disclosing sensitive information.
   {
     "status": 500,
     "error": "Server error",
-    "text": server_error // (e.g. exception text or sql error)
+    "text": "${server_error}" // (e.g. exception text or sql error)
```
Comment (mathuin) [inline source/errors.rst:58]: This is an excellent way to describe the diversity of potential strings.

File: source/errors.rst [hunk 13]
```diff
@@ -174,8 +176,6 @@ server is running.
     "text": "Invalid username or password" / "Bad oAuth token" / etc
```
Comment (mathuin) [inline source/errors.rst]: Do you want to do the same thing here you did with server error?  I think it would work well.

**Review by mathuin — APPROVED** [review]
I like it! All that's left is the possible change to the authentication failure description to match the server error. Otherwise, LGTM!

**Comment by mathuin** [comment]
LGTM!  Do you need additional reviewers?

**Comment by aaroncohen73** [comment]
@LyonesGamer Could you give this PR another look before I merge it?

**Review by MorganEPatch — APPROVED** [review]

**Pull request merged**